chia/data_layer/data_layer.py

```diff
@@ -482,9 +483,24 @@ async def add_missing_files(self, store_id: bytes32, overwrite: bool, foldername
             return
         max_generation = min(singleton_record.generation, 0 if root is None else root.generation)
         server_files_location = foldername if foldername is not None else self.server_files_location
+        files = []
         for generation in range(1, max_generation + 1):
             root = await self.data_store.get_tree_root(tree_id=store_id, generation=generation)
-            await write_files_for_root(self.data_store, store_id, root, server_files_location, overwrite)
+            res = await write_files_for_root(self.data_store, store_id, root, server_files_location, overwrite)
+            files.append(res.diff_tree.name)
+            files.append(res.full_tree.name)
+
+        uploaders = await self.get_uploaders(store_id)
+        if uploaders is not None and len(uploaders) > 0:
+            request_json = {"store_id": store_id.hex(), "files": json.dumps(files)}
+            for uploader in uploaders:
+                async with aiohttp.ClientSession() as session:
+                    async with session.post(uploader + "/add_missing_files", json=request_json) as response:
+                        res_json = await response.json()
+                        if not res_json["uploaded"]:
+                            self.log.error(f"failed to upload to uploader {uploader}")
+                        else:
+                            self.log.debug(f"uploaded to uploader {uploader}")
```

Comment:
If the intent is to hit all the uploaders, should we handle errors and continue?

Reply:
how do you suggest we do that besides just logging it and letting the user run this again ?

i think a retry mechanism is a bad idea

Reply:
Not retry, I think Kyle is asking about handling exceptions. Right now if an uploader fails to upload, the error is logged and the for loop will go to the next one. However, if the uploader is not running, the post will raise ClientError exception and the next uploader not be tried

Reply:
`upload_files` and `add_missing_files` are consistent in this, exceptions break the loop while other errors do not.
I would say the most common exception is an uploader not running so this will raise a ClientError and break out of the loop.

Reply:
Per internal discussion, created GH issue to track this: https://github.com/Chia-Network/chia-blockchain/issues/15081
and resolving conversation